client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormLegend from 'components/forms/form-legend';
+import FormRadio from 'components/forms/form-radio';
+import FormTextInput from 'components/forms/form-text-input';
+import PaymentMethodEditFormToggle from './payment-method-edit-form-toggle';
+
+class PaymentMethodStripe extends Component {
+
+	static propTypes = {
+		method: PropTypes.shape( {
+			settings: PropTypes.shape( {
+				title: PropTypes.shape( {
+					id: PropTypes.string.isRequired,
+					label: PropTypes.string.isRequired,
+					type: PropTypes.string.isRequired,
+					value: PropTypes.string.isRequired,
+				} ),
+			} ),
+		} ),
+		translate: PropTypes.func.isRequired,
+		onEditField: PropTypes.func.isRequired,
+	};
+
+	onEditFieldHandler = ( e ) => {
+		this.props.onEditField( e.target.name, e.target.value );
+	}
+
+	onSaveHandler = () => {
+		this.props.onSave( this.props.method );
+	}
+
+	renderEnabledField = ( isEnabled ) => {
+		return (
+			<PaymentMethodEditFormToggle
+				checked={ isEnabled === 'yes' ? true : false }
+				name="enabled"
+				onChange={ this.onEditFieldHandler } />
+		);
+	}
+
+	renderEditTextboxSecretKey = ( setting ) => {
+		const { translate } = this.props;
+		return (
+			<FormTextInput
+				name={ setting.id }
+				onChange={ this.onEditFieldHandler }
+				value={ setting.value }
+				placeholder={ translate( 'Enter your secret key from your Stripe.com account' ) } />
+		);
+	}
+
+	renderEditTextboxPublishableKey = ( setting ) => {
+		const { translate } = this.props;
+		return (
+			<FormTextInput
+				name={ setting.id }
+				onChange={ this.onEditFieldHandler }
+				value={ setting.value }
+				placeholder={ translate( 'Enter your publishable key from your Stripe.com account' ) } />
+		);
+	}
+
+	renderKeyFields = ( isLiveMode ) => {
+		const { method, translate } = this.props;
+		const testSecretLabel = translate( 'Test Secret Key' );
```

Comment:
To simplify the markup below (with the multiple if statements), maybe we could do something like:

`const secretLabel = isLiveMode ? translate( 'Live Secret Key' ) : translate( 'Test Secret Key' );`

`<FormLabel>{secretLabel}</FormLabel>`

etc..

That would also probably cut down on a few lines.